docs/trace-command.md

```diff
@@ -0,0 +1,117 @@
+wrong with accessing Kubernetes cluster, user needs to first find the corresponding `Resource Claim`, check its status
+and may need to continue with the `Managed Resource`, `Resource Class` and `Provider` resources.
+
+Trace command aims to ease debugging and troubleshooting process by providing a holistic view for the resource being 
+traced. 
+
+## Usage
+
+The trace tool is part of the crossplane kubectl plugin and can be used as follows:
+
+```bash
+$ kubectl crossplane trace --help
+Usage: kubectl crossplane trace TYPE[.GROUP] NAME [-n| --namespace NAMESPACE] [-h|--help]
+
+-h, --help                  Shows this help message
+    --kubeconfig string     Absolute path to the kubeconfig file
+-n, --namespace string      Namespace (default "default")
+-o, --outputFormat string   Output format. One of: dot
+```
+
+Examples:
+
+```
+### Text output
+# Trace a KubernetesApplication
+$ kubectl crossplane trace KubernetesApplication wordpress-app-83f04457-0b1b-4532-9691-f55cf6c0da6e -n app-project1-dev
+
+# Trace a MySQLInstance
+$ kubectl crossplane trace MySQLInstance wordpress-mysql-83f04457-0b1b-4532-9691-f55cf6c0da6e -n app-project1-dev
+
+### Graph output, which can be visualized with graphviz as follows.
+$ kubectl crossplane trace KubernetesApplication wordpress-app-83f04457-0b1b-4532-9691-f55cf6c0da6e -n app-project1-dev -o dot | dot -Tpng > /tmp/output.png
+```
+
+### State of a resource
+
+Each object in the trace output categorized in one of the following states which is represented in text (state column of
+overview) and graph output (as node attributes) accordingly.
+
+| State | Text Representation | Graph Representation | 
+| :--------: | :---: | :---: |
+| Ready    | ✔ | Solid - Black |
+| NotReady | ! | Solid - Orange |
+| Missing  | ✕ | Dotted - Red |
+
+### Sample outputs
+
+#### Text output 
+
+![alt text](media/trace-text-example.png "Trace Text output example")
+
+#### Graph output 
+
+![alt text](media/trace-graph-example.png "Trace Graph output example")
+
+
+## How it works
+
+The trace tool categorizes Crossplane resources based on a mapping between resource group/kinds and one of the following 
+Crossplane resource types. Then it finds the related resources with the help of the references available in the resource
+declarations as shown below. 
+
+- [Kubernetes Application](https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#kubernetes-application)
+    - `status.clusterRef` -> KubernetesCluster
+    - listing/filtering all `KubernetesApplicationResources` and `ResourceClaims` in the same namespace with labels 
+  at `spec.resourceSelector.matchLabels`
+  
+- Kubernetes Application Resource 
+    - `status.clusterRef` -> KubernetesCluster
+    - `spec.secrets` -> [Connection secrets](https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#connection-secrets)
+   
+- [Resource Claim](https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#kubernetes-application)
```

Comment:
I believe this link should be more like `https://github.com/crossplaneio/crossplane/blob/master/docs/concepts.md#resource-claim`